src/user/user.controller.ts

```diff
@@ -43,16 +44,24 @@ export class UserController {
     if (!req.user || !req.body.email) {
       throw new BadRequestException('Invalid user data.');
     }
-    return this.userService.findByEmail(req.body.email);
+    const user = await this.userService.findByEmail(req.body.email);
+    if (user == undefined) {
+      return { message: 'User was not found or has been deleted' };
+    }
+    return user;
   }
   
 
 @Get('profiles/:id')
   @UseGuards(JwtAuthGuard, RolesGuard)
   @Roles(Role.ADMIN)
   @UsePipes(new ValidationPipe({ transform: true }))
-  findOne(@Param() params: GetUserDto) {
-    return this.userService.findUserById(params.id);
+  async findOne(@Param() params: GetUserDto) {
```

Comment:
Is this endpoint really necessary? You should only be able to see your own profile

Reply:
fixed by https://github.com/alkemyTech/CENCO-MIT-Back-T1/pull/27/commits/200b86b0c16713c23653e196873ac7c0f5ad3ba3